pkg/monitoring/metrics/virt-controller/vmstats_collector.go

```diff
@@ -467,3 +479,50 @@ func containsCondition(target k6tv1.VirtualMachineConditionType, elems []k6tv1.V
 	}
 	return false
 }
+
+func CollectDiskAllocatedSize(vms []*k6tv1.VirtualMachine) []operatormetrics.CollectorResult {
+	var cr []operatormetrics.CollectorResult
+
+	for _, vm := range vms {
+		if vm.Spec.Template == nil {
+			continue
+		}
+		cr = append(cr, collectDiskMetricsFromPVC(vm)...)
+	}
+
+	log.Log.Infof("Returning result set: %+v", cr)
+	return cr
+}
+
+func collectDiskMetricsFromPVC(vm *k6tv1.VirtualMachine) []operatormetrics.CollectorResult {
+	var cr []operatormetrics.CollectorResult
+
+	for _, vol := range vm.Spec.Template.Spec.Volumes {
+		if vol.PersistentVolumeClaim != nil {
+			diskName := vol.Name
+			pvcName := vol.PersistentVolumeClaim.ClaimName
+
+			key := fmt.Sprintf("%s/%s", vm.Namespace, pvcName)
+			obj, exists, err := persistentVolumeClaimInformer.GetStore().GetByKey(key)
+			if err != nil || !exists {
+				log.Log.Warningf("Failed to retrieve PVC %s in namespace %s: %v", pvcName, vm.Namespace, err)
+				continue
+			}
+			pvc, ok := obj.(*k8sv1.PersistentVolumeClaim)
+			if ok {
+				pvcSize := pvc.Spec.Resources.Requests.Storage()
```

Comment:
Can you be more specific about what you mean by "allocated disk size"? I'm not sure this is the value you are looking for. This is the size of PVC that was requested. It does not map directly to the size of disk image used by the VM

Reply:
@mhenriks allocated size means the maximum size the disk can use

Reply:
Just a small correction. That is not the main use case.
Knowing the allocated size is important for many use cases like:
1. Resources capacity planning - See how much is allocated and plan for next year for example.
2. Resources utilization - We can calculate the used size out of the total
3. Resources management -  Free up unused storage